src/main/java/de/tum/in/www1/artemis/service/scheduled/NotificationScheduleService.java

```diff
@@ -112,4 +115,85 @@ private void scheduleNotificationForExercise(Exercise exercise) {
             log.error("Failed to schedule notification for exercise " + exercise.getId(), exception);
         }
     }
+
+    /// EXERCISE_SUBMISSION_ASSESSED
+
+    /**
+     * updateScheduling method for the notificationType EXERCISE_SUBMISSION_ASSESSED
+     * @param exercise that should trigger a notification when the assessment due date is over
+     */
+    public void updateSchedulingForAssessedExercisesSubmissions(Exercise exercise) {
+        checkSecurityUtils();
+        if (exercise.getAssessmentDueDate() == null || ZonedDateTime.now().isAfter(exercise.getAssessmentDueDate())) {
+            // to make sure no wrong notification is sent out the date is checked again in the concrete notification method
+            scheduleService.cancelScheduledTaskForLifecycle(exercise.getId(), ExerciseLifecycle.ASSESSMENT_DUE);
+            return;
+        }
+        if (exercise.isCourseExercise()) {
+            scheduleNotificationForAssessedExercisesSubmissions(exercise);
+        }
+    }
+
+    /**
+     * The place where the actual tasks/methods are called/scheduled that should be run when at the assessment due date of the exercise
+     * @param exercise which will be announced by a notifications at release date
+     */
+    private void scheduleNotificationForAssessedExercisesSubmissions(Exercise exercise) {
+        try {
+            checkSecurityUtils();
+            scheduleService.scheduleTask(exercise, ExerciseLifecycle.ASSESSMENT_DUE, () -> {
+                checkSecurityUtils();
+                Exercise foundCurrentVersionOfScheduledExercise = findCurrentVersionOfScheduledExercise(exercise);
+                if (foundCurrentVersionOfScheduledExercise == null) {
+                    return;
+                }
+                if (checkIfTimeIsCorrectForScheduledTask(foundCurrentVersionOfScheduledExercise.getAssessmentDueDate())) {
+                    singleUserNotificationService.notifyUsersAboutAssessedExerciseSubmission(foundCurrentVersionOfScheduledExercise);
+                }
+            });
+            log.debug("Scheduled notify about assessed exercise submission after assessment due date for exercise '{}' (#{}) at {}.", exercise.getTitle(), exercise.getId(),
+                    exercise.getAssessmentDueDate());
+        }
+        catch (Exception exception) {
+            log.error("Failed to schedule notification for exercise " + exercise.getId(), exception);
+        }
+    }
+
+    /**
+     * If the exercise has been updated in the meantime the scheduled immutable exercise is outdated and has to be replaced by the current one in the DB
+     *
+     * @param exercise that has to be checked
+     * @return the newest version of the exercise or null if none was found
+     */
+    private Exercise findCurrentVersionOfScheduledExercise(Exercise exercise) {
+        try {
```

Comment:
I would not try catch here, let the exception happen, this is an edge case that we should see in the logs

Reply:
I removed the try-catch block (and the method in general, because the other logic left in this method fits in one line)